Tests/ToolBox/ForexVolume/ForexVolumeDownloaderTest.cs

```diff
@@ -0,0 +1,342 @@
+﻿using Ionic.Zip;
+using NodaTime;
+using NUnit.Framework;
+using QuantConnect.Data;
+using QuantConnect.Data.Custom;
+using QuantConnect.ToolBox;
+using QuantConnect.ToolBox.FxVolumeDownloader;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace QuantConnect.Tests.ToolBox.FxVolume
+{
+    [SetUpFixture]
+    public class SetUpClass
+    {
+        [SetUp]
+        public void SetUpTests()
+        {
+            Market.Add("FXCMForexVolume", identifier: 20);
+        }
+    }
+
+    [TestFixture]
+    public class ForexVolumeDownloaderTest
+    {
+        [SetUp]
+        public void SetUpTemporatyFolder()
+        {
+            var randomFolder = "ForexVolumeTesting_" + Guid.NewGuid().ToString("N").Substring(startIndex: 0, length: 6);
+            _dataDirectory = Path.Combine(Path.GetTempPath(), randomFolder);
+            _downloader = new ForexVolumeDownloader(_dataDirectory);
```

Comment:
The idea here is to test if the data is written *as* I expect and *where* I expect. The data correctness is tested as you suggest in other tests.

I see as alternative, use the same testing assembly folder and make a temporary folder there, run the assert and delete the temporary folder after that. Is that any better? 

I'll use the Nunit tests cases features to shrink the testing file.